addons/web/controllers/main.py

```diff
@@ -629,14 +583,27 @@ def _concat_xml(self, file_dict):
 
         return etree.tostring(root, encoding='utf-8') if root is not None else b'', checksum.hexdigest()[:64]
 
+    def _get_asset_paths(self):
+        """Proxy for ir_asset.get_asset_paths
+        Useful to make 'self' testable.
+        """
+        return request.env['ir.asset'].get_asset_paths(addons=self.addons, bundle='web.assets_qweb', xml=True)
+
     def _get_qweb_templates(self):
         """One and only entry point that gets and evaluates static qweb templates
 
         :rtype: (str, str)
         """
-        files = OrderedDict([(addon, list()) for addon in self.addons])
-        [files[f[2]].append(f[0]) for f in self._manifest_glob()]
-        content, checksum = self._concat_xml(files)
+        xml_paths = OrderedDict()
```

Comment:
From Python 3.6 on, all dicts are de facto ordered.  So `xml_paths = {}` works just as good.
Actually, `OrderedDict` is only useful if you want to reorder keys (move to first, move to last).